### ISSUE TITLE
`no-unsafe-string-replacement`: False positive on non-String `.replace()` methods

### ISSUE THREAD

**Issue by nozomiishii** [sindresorhus/eslint-plugin-unicorn#3437]
First of all, thank you for maintaining this amazing plugin — it's one of my favorite ESLint plugins and I use it across all my projects. Really appreciate the work you put into it! 🦄

## Bug

The `no-unsafe-string-replacement` rule flags any `.replace()` call that matches the AST shape (method name + 2 arguments + non-literal second argument), regardless of whether the receiver is actually a `String`.

This causes false positives on APIs like `next/navigation`'s `router.replace()`, which is a navigation method, not `String#replace()`.

## Rule

`no-unsafe-string-replacement`

## Example

```js
/* eslint unicorn/no-unsafe-string-replacement: "error" */

import { useRouter } from "next/navigation";

const router = useRouter();
const pathname = "/about";
const locale = "en";

// ⚠️ Flagged: "Do not use a non-literal replacement value with String#replace()"
// This is Router#replace(), not String#replace()
router.replace(pathname, { locale });
```

The second argument is an `ObjectExpression`, which is not a valid argument for `String#replace()` (only `string` or `function`). The rule should not flag this case.

## Root cause

`isMethodCall` in the rule implementation has no `object` constraint, so it matches `anything.replace(...)`. The rule only checks:

1. Method name is `replace` or `replaceAll`
2. Exactly 2 arguments
3. Second argument is not a string literal or function

It does not check whether the receiver is a `string`.

## Environment

- `eslint-plugin-unicorn` v68.0.0
- ESLint v10.3.0
- Node.js v24
